src/sst.rs

```diff
@@ -431,7 +446,11 @@ impl<'a> EncodedSsTableBuilder<'a> {
             let filter = Arc::new(self.filter_builder.build());
             let encoded_filter = filter.encode();
             filter_len = encoded_filter.len();
```

Comment:
I believe this is the issue. The filter length is set to the encoded bytes _before_ they're compressed. The length should be set _after_ compression. I moved this line above the linen 453 (`buf.put(...)`) and the tests passed.

Reply:
ah thanks